subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/DaemonGradleExecuter.java

```diff
@@ -29,21 +26,34 @@
 import static org.apache.commons.collections.CollectionUtils.containsAny;
 
 public class DaemonGradleExecuter extends NoDaemonGradleExecuter {
-    private static final JvmVersionDetector JVM_VERSION_DETECTOR = GLOBAL_SERVICES.get(JvmVersionDetector.class);
+
+    private boolean daemonExplicitlyRequired;
```

Comment:
The fact that `InProcessGradleExecuter` calls `requireDaemon()` but does not actually require a daemon is fairly confusing.  I wonder if it might be less convoluted if we kept the old behavior of `requireDaemon()` but used a different method in the `DaemonGradleExecuter` constructor like `applyDaemonRequirements()` which `InProcessGradleExecuter` could just override and not call `requireDaemon()`?

Fundamentally, the relationship here feels not great.  In other words, an `InProcessGradleExecuter` is not _really_ a `DaemonGradleExecuter`, but it behaves like one in certain cases.  It seems like this should be more compositional (e.g. he just creates a `DaemonGradleExecuter` to do the work for him when forking is required).  I don't think we need to change anything now, but it would be great if we could continue to evolve this into something more conceptually consistent.

Reply:
Yeah, composing the two makes more sense. I just wanted to make the minimal change to make this faster.